python/ray/tune/utils/util.py

```diff
@@ -231,6 +231,18 @@ def flatten_dict(dt, delimiter="/"):
     return dt
 
 
+def unflatten_dict(dt, delimiter="/"):
+    """Unflatten dict. Does not support unflattening lists."""
+    out = defaultdict(dict)
+    for key, val in dt.items():
+        path = key.split(delimiter)
+        item = out
+        for k in path[:-1]:
+            item = item[k]
```

Comment:
can't this be shortened to `item = item[path[-2]]`? I think I'm misunderstanding something here

Reply:
Path here is usually a tuple like `(a, b, c)`. We would like to set `dt[a][b][c]=x` and do this by setting `item=dt`, then `item = item[a]`, then `item=item[b]` and finally `item[c]=x`. Does that make sense?
See also `assign_value()` in `variant_generator.py`